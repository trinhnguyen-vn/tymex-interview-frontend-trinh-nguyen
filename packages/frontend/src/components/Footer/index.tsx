"use client"
import React from 'react';
import {
    Box,
    Typography,
    TextField,
    Button,
    Grid
} from '@mui/material';
import PhoneIcon from '@mui/icons-material/Phone';
import EmailIcon from '@mui/icons-material/Email';
import { MAIN_BACKGROUND_COLOR } from '@/styles';
import { StyledLink } from '@/styles/Link';
import { NAV_LINKS } from '@/constants/navbar';

const SECURITY_LINKS = [{ path: '/', label: "Security" }, { path: '/', label: "Legal" }, { path: '/', label: "Privacy" }]
const OTHER_LINKS = [{ path: '/faqs', label: "FAQs" }, { path: '/', label: "News" }, { path: '/', label: "Community" }]
const FOOTER_LINKS = [...NAV_LINKS, ...OTHER_LINKS]

const Footer: React.FC = () => {
    const handleSubscribe = (event: React.FormEvent) => {
        event.preventDefault();
        // Add subscription logic here
        console.log('Subscribed with:');
    };

    return (
        <Box
            sx={{
                background: MAIN_BACKGROUND_COLOR,
                color: '#fff',
                py: 4,
                px: 5,
            }}
        >
            <Grid container spacing={3} justifyContent="space-between">
                {/* Navigation Section */}
                <Grid size={{ xs: 12, md: 3, sm: 4 }}>
                    <Typography variant="h6" gutterBottom sx={{ fontWeight: 'bold' }}>
                        NAVIGATION
                    </Typography>
                    <Box sx={{
                        display: 'grid',
                        gridTemplateColumns: '1fr 1fr 1fr', gap: "16px 32px", width: 300
                    }}>
                        {FOOTER_LINKS.map(item => (
                            <StyledLink href={item.path} passHref key={item.label}>
                                {item.label}
                            </StyledLink>
                        ))}
                    </Box>
                </Grid>

                {/* Contact Us Section */}
                <Grid size={{ xs: 12, md: 2, sm: 4 }}>
                    <Typography variant="h6" gutterBottom sx={{ fontWeight: 'bold' }}>
                        CONTACT US
                    </Typography>
                    <Box sx={{ display: 'flex', alignItems: 'center', my: 2 }}>
                        <PhoneIcon sx={{ mr: 1 }} />
                        <Typography variant="body2">01234568910</Typography>
                    </Box>
                    <Box sx={{ display: 'flex', alignItems: 'center', my: 2 }}>
                        <EmailIcon sx={{ mr: 1 }} />
                        <Typography variant="body2">tymex-talent@tyme.com</Typography>
                    </Box>
                </Grid>

                {/* Subscribe Section */}
                <Grid size={{ xs: 12, md: 3, sm: 4 }}>
                    <Typography variant="h6" gutterBottom sx={{ fontWeight: 'bold' }}>
                        SUBSCRIBE TO RECEIVE OUR LATEST UPDATE
                    </Typography>
                    <form onSubmit={handleSubscribe}>
                        <TextField
                            fullWidth
                            variant="outlined"
                            placeholder="Your email address"
                            sx={{
                                backgroundColor: 'background.default',
                                borderRadius: 1,
                                mb: 2,
                            }}
                            inputProps={{ style: { color: '#000' } }}
                        />
                        <Button
                            type="submit"
                            variant="contained"
                        >
                            Subscribe
                        </Button>
                    </form>
                </Grid>
            </Grid>

            <Grid container justifyContent="space-between"
                sx={{ mt: 4, textAlign: 'center', borderTop: '1px solid #333', pt: 2 }}
            >
                <Typography variant="body2" sx={{ color: '#fff', mt: 2 }}>
                    Â©2023 Tyme - Edit. All Rights reserved.
                </Typography>
                <Box sx={{ mt: 2, display: 'flex', gap: 2 }}>
                    {SECURITY_LINKS.map(item => (
                        <StyledLink key={item.label} href={item.path} passHref >{item.label}</StyledLink>
                    ))}
                </Box>
            </Grid>
        </Box>
    );
};

export default Footer;