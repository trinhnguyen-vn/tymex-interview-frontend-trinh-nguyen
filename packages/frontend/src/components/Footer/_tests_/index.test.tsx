import { render } from '@testing-library/react';
import Footer from '..';

it('renders Footer correctly', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
});