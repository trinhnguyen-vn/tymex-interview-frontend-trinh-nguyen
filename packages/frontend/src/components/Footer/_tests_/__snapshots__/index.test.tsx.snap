// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-16jkopy"
  >
    <div
      class="MuiBox-root css-19cbedo"
    />
    <div
      class="MuiBox-root css-1aphctg"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-3 css-2b9518-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 MuiGrid-grid-md-3 MuiGrid-grid-sm-4 css-gxw8zd-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1jrq8e9-MuiTypography-root"
          >
            NAVIGATION
          </h6>
          <div
            class="MuiBox-root css-cuyma4"
          >
            <a
              class="css-g70xym"
              href="/"
            >
              Home
            </a>
            <a
              class="css-g70xym"
              href="/about"
            >
              About us
            </a>
            <a
              class="css-g70xym"
              href="/teams"
            >
              Our teams
            </a>
            <a
              class="css-g70xym"
              href="/marketplace"
            >
              Marketplace
            </a>
            <a
              class="css-g70xym"
              href="/roadmap"
            >
              Roadmap
            </a>
            <a
              class="css-g70xym"
              href="/whitepaper"
            >
              Whitepaper
            </a>
            <a
              class="css-g70xym"
              href="/faqs"
            >
              FAQs
            </a>
            <a
              class="css-g70xym"
              href="/"
            >
              News
            </a>
            <a
              class="css-g70xym"
              href="/"
            >
              Community
            </a>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 MuiGrid-grid-md-2 MuiGrid-grid-sm-4 css-1hjfhws-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1jrq8e9-MuiTypography-root"
          >
            CONTACT US
          </h6>
          <div
            class="MuiBox-root css-rjvdxb"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-15a76qm-MuiSvgIcon-root"
              data-testid="PhoneIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
            >
              01234568910
            </p>
          </div>
          <div
            class="MuiBox-root css-rjvdxb"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-15a76qm-MuiSvgIcon-root"
              data-testid="EmailIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 4-8 5-8-5V6l8 5 8-5z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
            >
              tymex-talent@tyme.com
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 MuiGrid-grid-md-3 MuiGrid-grid-sm-4 css-gxw8zd-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1jrq8e9-MuiTypography-root"
          >
            SUBSCRIBE TO RECEIVE OUR LATEST UPDATE
          </h6>
          <form>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-13r7ndo-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                  id="«r0»"
                  placeholder="Your email address"
                  style="color: rgb(0, 0, 0);"
                  type="text"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-w4cd9x"
                  >
                    <span
                      aria-hidden="true"
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Subscribe
            </button>
          </form>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-1xmcgsq-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-4275by-MuiTypography-root"
        >
          ©2023 Tyme - Edit. All Rights reserved.
        </p>
        <div
          class="MuiBox-root css-fbr47u"
        >
          <a
            class="css-g70xym"
            href="/"
          >
            Security
          </a>
          <a
            class="css-g70xym"
            href="/"
          >
            Legal
          </a>
          <a
            class="css-g70xym"
            href="/"
          >
            Privacy
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
